### ISSUE TITLE
@media "screen" queries aren't matched

### ISSUE THREAD

**Issue by vgmoose** [litehtml/litehtml#359]
Thank you for this project!

I'm logging this issue to report what I think is a regression with newer media query css changes: @media query `screen` CSS seem to be ignored. This can be noticed when rendering Wikipedia's home page.

According to the linked [media queries doc](https://drafts.csswg.org/mediaqueries/#media-types), `screen` should be:
> Matches all devices that aren’t matched by [print](https://drafts.csswg.org/mediaqueries/#valdef-media-print).

And `print` is:
> Matches printers, and devices intended to reproduce a printed display, such as a web browser showing a document in “Print Preview”.

To confirm the behavior there, if you try to print the Wikipedia home page in Chrome/Safari, the blue/green background colors disappear in the print preview. So the intent by Wikipedia is that these backgrounds are only rendered on device, but not when printing the page.

Current master branch:
<img src="https://github.com/user-attachments/assets/f24b1098-100c-45a1-9f5c-6b7e139a31db" alt="current master branch" width=600 />

\
Hardcoding a match for `media_type_screen` in [media_query.cpp](https://github.com/litehtml/litehtml/blob/master/src/media_query.cpp#L123):
<img src="https://github.com/user-attachments/assets/f73fcc39-4791-4f30-81eb-24ca09dbd12a" alt="screenshot with media screen" width=600 />

The CSS on the wikipedia page looks like:
```css
@media screen {
    #mp-left {
        background-color: #f5fffa;
        border-color: #cef2e0;
    }
    .mw-parser-output #mp-left .mp-h2 {
        background-color: #cef2e0;
    }
}
```

Can also replicate it with the following test page, which doesn't render any of the styling:
```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Media Query Test Page</title>
    <style>
        @media screen {
            div {
                width: 200px;
                height: 200px;
                font-family: sans-serif;
                margin: 10px;
            }
            .red {
                background-color:red;
            }
            .blue {
                background-color:blue;
            }
        }
    </style>
</head>
<body>
    <div class="red">Red</div>
    <div class="blue">Blue</div>
</body>
</html>
```

**Comment by casesolved-co-uk** [comment]
Out of interest (sorry slightly off-topic) are @media queries supported in general,
e.g. `@media (min-width: 768px) {` for responsive layouts?

**Comment by vgmoose** [comment]
They are supported, for instance the below will render the background colors only if the window width is at least 768px wide. The width and other attributes are reported by `get_media_features` in the class that implements [document_container](https://github.com/litehtml/litehtml/wiki/document_container#get_media_features). And the fields to fill out in that method for what values should be expected there are in [`media_features`](https://github.com/litehtml/litehtml/blob/dc857e2702e462609add1985fbc609bc0a638410/include/litehtml/types.h#L925) struct.

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Media Query Test Page</title>
    <style>
        div {
            width: 200px;
            height: 200px;
            font-family: sans-serif;
            margin: 10px;
        }
        @media (min-width:768px) {
            .red {
                background-color:red;
            }
            .blue {
                background-color:blue;
            }
        }
    </style>
</head>
<body>
    <div class="red">Red</div>
    <div class="blue">Blue</div>
</body>
</html>
```

**Comment by casesolved-co-uk** [comment]
@vgmoose okay thanks. Was contemplating using this library for an embedded project which already needs to serve a web page UI. Then wondered if I could use this library as a client to render the same UI on a tiny screen; why design twice? Sounds like it's possible...

**Comment by vgmoose** [comment]
It works pretty well for displaying HTML and CSS! I recommend checking your specific page in [litebrowser-linux](https://github.com/litehtml/litebrowser-linux) to see how well it renders and then go from there.

**Comment by casesolved-co-uk** [comment]
Oh no I mean compiling it for ESP32 and freertos

**Comment by vgmoose** [comment]
Yeah! But, I meant as a test, to make sure it renders the kinds of components on your target page first, before investing the time into a litehtml port. It's not too bad to integrate, but without javascript many web pages don't display properly (or at all).

**Comment by vgmoose** [comment]
Opened a PR for the `screen` media query issue. I think that there may still be scenarios where someone wants to use litehtml to render a print preview, for instance. But the general case of supporting screens seems more common, and one or the other should be matched to be compliant.

**Comment by tordex** [comment]
@vgmoose, sorry, but what your PR does? As I see it just force applying all ```@media screen```. This is incorrect.

Also I don't understand the issue. I've tried your examples, all of them are rendered correctly in the [litebrowser-linux](https://github.com/litehtml/litebrowser-linux).

**Comment by vgmoose** [comment]
Indeed it was a bug on my end, and hardcoding the media type to screen is wrong. I put more info in the PR, if others are interested. Thank you!

**Comment by casesolved-co-uk** [comment]
> Yeah! But, I meant as a test, to make sure it renders the kinds of components on your target page first, before investing the time into a litehtml port. It's not too bad to integrate, but without javascript many web pages don't display properly (or at all).

Ah I see. Yes I might try that when I've written the web page, thanks! To be fair I'm now thinking of using vue.js so probably not but simpler static web pages with MVP.css for example could be interesting.
